### ISSUE TITLE
Resources in subfolders not found

### ISSUE THREAD

**Issue by busybeaver** [waysact/webpack-subresource-integrity#9]
It seems like the plugin has problems with resources located in subfolders.

My webpack output dir looks something like:

```
dist
├─styles
│  ├──*.css
├─images
│  ├──...
├─js
│  ├──*.js
└─index.html
```

Using the 'webpack-subresource-integrity' (in combination with the html-webpack-plugin), the following warning occurs for all resources:

> WARNING in webpack-subresource-integrity: cannot determine hash for asset 'js/entry.js', the resource will be unprotected.

After some debugging, the issue seems to be in [index.js#L164](/waysact/webpack-subresource-integrity/blob/f6aae779f6c88f9316e81bacb6cf91ab5bba948c/index.js#L164). `compilation.assets` stores all assets with path, the `src` parameter includes the path as well, but then `path.basename(src)` removes the path. As a result, the property lookup in `compilation.assets` fails (i.e. `compilation.assets[path.basename(src)]` doesn't return a asset).

![image](https://cloud.githubusercontent.com/assets/858259/17508951/d11c05fa-5e17-11e6-8043-17c7191e9d48.png)

**Comment by jscheid** [comment]
@SPSpwetter this is fixed by simply not calling `basename`. Do you remember why you've added it in the first place? Do you think this change could break anything?

I've also removed the `basename` call from your example in the README and while doing so I've noticed that the example needlessly checks whether the integrity value is available. As far as I can tell, there should always be an integrity hash for every chunk script. Do you see anything wrong with just removing the conditional in the example?

**Comment by SPSpwetter** [comment]
I added it because the compilation.assets keys seemed to be just base name, while the chunk key was a longer path. That is, compilation.assets[src] didn't work.

**Comment by SPSpwetter** [comment]
As far as the conditional goes, I was just paranoid about assets that weren't processed.

**Comment by jscheid** [comment]
In my test cases the source filename and the asset key are identical (they are both relative to the the root directory). Could I ask you to try and recreate the situation in which they were different?

**Comment by SPSpwetter** [comment]
You mean run my regular build? Sure... Is there a trick for getting console.log to show before/after? I think the problem is that my html is in a separate path, so I have to generate ../../html/home.html

**Comment by jscheid** [comment]
@SPSpwetter perfect, I can reproduce with that information. I'll take it from here, thanks!

**Comment by jscheid** [comment]
@SPSpwetter should cover that case too now. Any chance you could give my branch a try in your application build?

**Comment by SPSpwetter** [comment]
Yeah I can try that.

**Comment by SPSpwetter** [comment]
Thumbs up. Let Go The Merge.

**Comment by jscheid** [comment]
Thanks all, released in v0.5.1 🎉

**Comment by busybeaver** [comment]
thanks everyone for the quick resolution 👍
